### PR TITLE
Optimizing hyper params

### PR DESCRIPTION
**Pull request by KYNIKER**
LAver Fredriks arbejde.

### CODE REVIEW

File: Main/Bayesian_Search.py
Comment (ShankzHD) [inline Main/Bayesian_Search.py:1]: Create separate file for helper functions 

Datasplit should be a parameter

File: Main/Search_Model.py
Comment (ShankzHD) [inline Main/Search_Model.py:1]: gridModel not good name
search_model maybe not good name

File: Main/Classes.py [hunk 1]
```diff
@@ -72,10 +87,73 @@ class TestData():
...
+        self.test_scores = []       #Model Guesses
+        self.test_labels = []       #Truths
         self.test_probability_estimates = []
 
+# Data covering different evaluation metrics based on input TestData
+class EvaluationMetricsData():
+    def __init__(self, TestData):
+        self.accuracy = TestData.test_accuracy
+        self.TP = 0
+        self.TF = 0
```
Comment (ShankzHD) [inline Main/Classes.py:99]: true false? TN??

File: Main/Grid_Search.py [hunk 2]
```diff
@@ -1,67 +1,78 @@
...
 from tqdm import trange
+from torch_geometric.loader import DataLoader
 
 from sklearn.model_selection import train_test_split
 
-
-# constants
-EPOCHS = 300
-
+DATASPLIT = 150
```
Comment (ShankzHD) [inline Main/Grid_Search.py:12]: PARAM

File: Main/Grid_Search.py [hunk 3]
```diff
@@ -1,67 +1,78 @@
...
-def grid_search (train_loader, test_loader, device, param_grid):
+def grid_search (dataset, device, param_grid):
     # Initialize variables to keep track of the best model and its accuracy
-    best_model = None
-    best_epoch_accuracy = 0.0
-    best_params = None
+    # Testing for test accuracy, f1 score, auc-roc, and auc-pr
+    
+    #StoredModel for each of the evalutationMetrics
+    best_accuracy = StoredModel()
```
Comment (ShankzHD) [inline Main/Grid_Search.py:25]: best_model_accuracy

File: Main/Grid_Search.py [hunk 4]
```diff
@@ -1,67 +1,78 @@
...
+    best_f1 = StoredModel()
```
Comment (ShankzHD) [inline Main/Grid_Search.py:26]: best_model_f1

File: Main/Grid_Search.py [hunk 5]
```diff
@@ -1,67 +1,78 @@
...
+    best_roc = StoredModel()
```
Comment (ShankzHD) [inline Main/Grid_Search.py:27]: best_model_roc

File: Main/Grid_Search.py [hunk 6]
```diff
@@ -1,67 +1,78 @@
...
+    best_pr = StoredModel()
```
Comment (ShankzHD) [inline Main/Grid_Search.py:28]: best_model_pr

File: Main/Grid_Search.py [hunk 7]
```diff
@@ -1,67 +1,78 @@
...
         'dropout_rate': [0.25, 0.5, 0.75],
         'hidden_channels': [32, 64],
         'learning_rate': [0.1, 0.01, 0.005],
+        'batch_size' : [1, 32, 64],
+        'epochs' : [150, 300],
+        'amount_of_layers' : [3,4,5],
+        'optimizer' : ['adam', 'sgd'],
+        'activation_function' : ['relu', 'sigmoid']
+        'pooling_algorithm' : ['mean', 'sum', 'max']
     } """
```
Comment (ShankzHD) [inline Main/Grid_Search.py:40]: Delete

File: Main/Random_Search.py
Comment (ShankzHD) [inline Main/Random_Search.py:1]: Delete

File: Main/Search_Model.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+        #Deciding which optimizer to use
+        optimizer = None
+        if(params["optimizer"].lower() == 'sgd'): optimizer = torch.optim.SGD(gridModel.parameters(), lr=gridModel.learning_rate)
+        elif(params["optimizer"].lower() == 'adam'): optimizer = torch.optim.Adam(gridModel.parameters(), lr=gridModel.learning_rate)
+        elif(params["optimizer"].lower() == 'rmsprop'): optimizer = torch.optim.RMSprop(gridModel.parameters(), lr=gridModel.learning_rate)
+        else : raise Exception("Invalid optimizer name: " + str(params["optimizer"]))
+
+        loss_function = torch.nn.CrossEntropyLoss()
+        baseModel = BaseModel(gridModel, loss_function, optimizer)
```
Comment (ShankzHD) [inline Main/Search_Model.py:33]: ^Should be its own function (create_model)

File: Main/Classes.py (hunk 1 above)
Comment (Grace-Melchiors) [reply Main/Classes.py:99]: sure go for it

File: Main/Grid_Search.py (hunk 2 above)
Comment (Grace-Melchiors) [reply Main/Grid_Search.py:12]: sure, just change it

File: Main/Grid_Search.py (hunk 3 above)
Comment (Grace-Melchiors) [reply Main/Grid_Search.py:25]: sure change it

File: Main/Grid_Search.py (hunk 5 above)
Comment (Grace-Melchiors) [reply Main/Grid_Search.py:27]: change it

File: Main/Grid_Search.py (hunk 6 above)
Comment (Grace-Melchiors) [reply Main/Grid_Search.py:28]: refer to above statement

File: Main/Random_Search.py
Comment (Grace-Melchiors) [reply Main/Random_Search.py:1]: go for it

File: Main/Bayesian_Search.py
Comment (Grace-Melchiors) [reply Main/Bayesian_Search.py:1]: well, let's discuss whether we want it as a a parameter first maybe

In terms of seperate file for helper functions you go ahead

File: Main/Search_Model.py
Comment (Grace-Melchiors) [reply Main/Search_Model.py:1]: sure change them

File: Main/Search_Model.py (hunk 8 above)
Comment (Grace-Melchiors) [reply Main/Search_Model.py:33]: you can do that sure

**Pull request merged**